roles/confluent.ksql/tasks/main.yml

```
@@ -47,6 +51,17 @@
     state: directory
     mode: 0755
 
+# Archive File deployments need to create SystemD service units
+# Copy the tarball's systemd service to the system
+- name: Copy Ksql Service from archive file to system
+  copy:
+    src: "{{binary_base_path}}/lib/systemd/system/confluent-ksqldb.service"
```

this one makes me nervous, bc we have logic elsewhere between ksql and ksqldb. One solution would be to use the variable that has the ternary, and use the basename filter
```suggestion
    src: "{{binary_base_path}}/lib/systemd/system/{{ksql.systemd_file|basename}}"
```

But if you change it here, should it be changed everywhere?

---

Great catch! I changed this to `src: "{{binary_base_path}}/lib/systemd/system/{{ksql_service_name}}.service"`

---

if you are changing it here, i think itd be best to change it everywhere?

---

Yes, that is correct, I changed it everywhere and am testing with molecule now.